### PR TITLE
Add SigLIP

### PR DESCRIPTION
**Pull request by james77777778**
This PR mainly adds `SigLIPBackbone`

You can check the colab for numerical results:
https://colab.research.google.com/drive/1R99yFJCbxTEpcxFHa2RtlwQWahUIPCJC?usp=sharing

|Preset|Library|Probability|
|-|-|-|
|siglip_base_patch16_224|`transformers`|19.8% (cats); 0.0% (dogs)|
|siglip_base_patch16_224|`keras_hub`|18.9% (cats); 0.0%(dogs)|
|siglip_so400m_patch14_384|`transformers`|50.9% (cats); 0.0% (dogs)|
|siglip_so400m_patch14_384|`keras_hub`|51.0% (cats); 0.0%(dogs)|

~~Let me know if this looks good to you. If so, I will add more presets.~~
The model weights have been uploaded here:
https://www.kaggle.com/models/kerashub/siglip/

Notes:
- We need to modify `keras.layers.Resizing` to configure `antialias`. Otherwise, the numerical results will differ a lot (e.g., 18.9% -> 13.5%)
- If I'm avaiable, I will look into SigLIP2
- `fit` is not supported yet but adding it should be straightforward

### CODE REVIEW

**Review by divyashreepathihalli — APPROVED** [review]

**Comment by james77777778** [comment]
The failure of the JAX GPU CI should be irrelevant to this PR

**Comment by divyashreepathihalli** [comment]
rerunning tests - have you uploaded the weights anywhere

**Comment by james77777778** [comment]
> rerunning tests - have you uploaded the weights anywhere

I have uploaded all available model weights here:
https://www.kaggle.com/models/kerashub/siglip/keras/

The presets have also been updated

**Comment by divyashreepathihalli** [comment]
quick question - why is fit not supported?

**Comment by james77777778** [comment]
> quick question - why is fit not supported?

I was a bit lazy adding it before 😅
Now, `SigLIP` supports `fit` and the numerics match HF's impl.

https://colab.research.google.com/drive/1R99yFJCbxTEpcxFHa2RtlwQWahUIPCJC?usp=sharing

|Preset|Library|Probability|Loss|
|-|-|-|-|
|siglip_base_patch16_224|`transformers`|19.8% (cats); 0.0% (dogs)|6.026|
|siglip_base_patch16_224|`keras_hub`|18.9% (cats); 0.0%(dogs)|6.061|
|siglip_so400m_patch14_384|`transformers`|50.9% (cats); 0.0% (dogs)|5.500|
|siglip_so400m_patch14_384|`keras_hub`|51.0% (cats); 0.0%(dogs)|5.511|
